### ISSUE TITLE
Grab tweets from before bot was initially started

### ISSUE THREAD

**Issue by Casuallynoted** [robertoszek/pleroma-twitter-info-grabber#32]
Hello! I was wondering, is there any possibility of an option for the bot to mirror over the tweets (as a one-time thing) from before it was run the first time? This way the pleroma profile could be a complete recreation of the twitter profile it's mirroring.

**Comment by robertoszek** [comment]
Hi, yeah, the first run experience isn't as smooth as I'd like.
As you mentioned, maybe the bot could try to detect if it's being executed for the first time (checking for the existence of residual files in cwd, perhaps, or if the Fediverse account has no toots/posts) and present the user the option of how it should proceed. Something along the lines of:

```
It seems like pleroma-bot is running for the first time. 
How far back should we retrieve tweets from the Twitter account? 

Enter a date (YYYY-MM-DD):
[Leave it empty to retrieve *ALL* tweets or enter 'continue' if you want the bot to execute as normal (checking date of last post in the Fediverse account)]
```

Would that suffice?

I need to look at Twitter's API docs again because v2 changed some things about fetching tweets older than a week.

**Comment by Casuallynoted** [comment]
That sounds awesome tbh! Maybe a modifier to have it ask that if the bot has already been run before too!

**Comment by robertoszek** [comment]
Absolutely, I agree.
```console
$ pleroma-bot forceDate [twitter_username]
```
Or something similar. The ```twitter_username``` would be optional, in case you have multiple users on your config and you only want to force the retrieval for one of them.

It would basically print the same thing excluding the first run bit:
```
How far back should we retrieve tweets
from the Twitter account? 

Enter a date (YYYY-MM-DD):
[Leave it empty to retrieve *ALL* tweets or
enter 'continue' if you want the bot to
execute as normal (checking date of last
post in the Fediverse account)]
```

**Comment by Casuallynoted** [comment]
That is so awesome 😄 That would be perfect! Thank you so much 😀

**Comment by robertoszek** [comment]
Hey,

I've implemented these changes here a72f888db888c8aa419333b41307f5c2b5776bdd

I have to improve the code coverage so it passes the tests (it dropped from 100% to 97%) before I integrate them into the master branch and do a proper release but in the meantime I've uploaded it as an experimental release on TestPyPI.

You can install it like this:
```console
$ pip install --index-url https://test.pypi.org/simple/ pleroma-bot --upgrade
```
Version number should be ```0.6.9``` after the upgrade.
The usage is as follows:
```
optional arguments:
  -h, --help            show this help message and exit
  -n, --noProfile       skips Fediverse profile update (no background image,
                        profile image, bio text, etc.)
  --forceDate [FORCEDATE]
                        forces the tweet retrieval to start from a specific
                        date. The twitter_username value (FORCEDATE) can be
                        supplied to only force it for that particular user in
                        the config
  -s, --skipChecks      skips first run checks
  --version             show program's version number and exit
```

So, for example:
```console
$ pleroma-bot --forceDate myTwitterUser
```
Should prompt you to enter a date when processing ```myTwitterUser```, let me know if it works in your case.
Cheers!

**Comment by robertoszek** [comment]
Hi,
The changes have been roll out to master and a new release has been published (```v0.7.0```).
You can upgrade as usual:

```console
$ pip install pleroma-bot --upgrade
```

Let me know if you encounter any issues with this new version, cheers!
